### ISSUE TITLE
React Native Drawer flashes after keyboard closes

### ISSUE THREAD

**Issue by CursedMun** [react-navigation/react-navigation#10715]
### Current behavior

I faced a strange problem that happened only on a small screen, I tested on 480x800 4 inches and 2400x1080 6.6inches screen, and that happened on the smaller one (480x800), the drawer flashes on any window that contains a text input after the keyboard closes.

Just for understanding better the problem [there's a video]( https://drive.google.com/file/d/1UuFFSLAqkMzjGumackKjkqk_UymG4iML/view?usp=sharing)
### Expected behavior

I would like it not to flash

### Reproduction

https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbwFBzgYygUwIY0wESmwHdMoA5bAN2AHNdoAaFOQksgYQgDs9eBlDBAA2wgGrBMxZqjakoASTwhmAXzgAzKBBBwA5AAEs2NDAC03anVzAeAegAmReXqRoeAZ3hyycALzoxng+FFb0MNAAFACUSAA8IQB0lDTh0CyoqA6YMCYAFgqWpsBUmIJYmNwefgga2MIemKoZmQCujQAymPRoAJ4K4MKYIJW5MLbcNXUNTS2owNzA4-UAShCteJQjfgBEABI6mDtzcE7sUFy8ozWRYNpgHtH+AHyIJ6hYMK1Q3HCR75kEs5ODw+DByiJxJJiIhEnC7hAHqpngDMnAgeclMNUWi4MB3JMEDEXm9cWTcZ9vr9-uTaYCFDx8Z4cXS4MI6HkYCy6ZZtjtsFBtMQzAAjEwAazM6xg7O4R25tI8wAAXpgagAmAAMzVZrPcwmguxFwglx115LsKPNaNi1rgqh11pNIswwl2AFlKq0zdaeAAFLAeapEp5+K12uAIh6JSypGw8RIOYAeMC4NB5SIhACCxU8iTQBsaIRitutjtZlu5cQkUgVcC8vSGNWQEbgIug2SgACEIDAIiAuAaoAAufQAYgAnFPp3oZBH21BOz2+zoAOrABwwPKjzWJACsc-NDu5ldZcTsIUuYIhohrxHDZNLaOPZI8GEwlQA8mBxp4bqHXhbclKR+Uk6Q8YhgDATAAFFLGNTAHFHGAoFaTBD3JPIcE7fg8ggYhuFHaZGgw58ThfTIH0BJJyg-X5eTVfZDh2dAdEgOVeBqA4RnUU9AQvYEoGSMIGCgZ4gA

### Platform

- [X] Android
- [ ] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/​bottom-tabs
- [X] @react-navigation/​drawer
- [ ] @react-navigation/​material-bottom-tabs
- [ ] @react-navigation/​material-top-tabs
- [ ] @react-navigation/​stack
- [ ] @react-navigation/​native-stack

### Environment

- [] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/drawer               |  6.1.8
| @react-navigation/native               |  6.0.10

**Comment by CursedMun** [comment]
> The versions mentioned in the issue for the following packages differ from the latest versions on npm:
> 
> * `@react-navigation/drawer` (found: `6.1.8`, latest: `6.4.3`)
> * `@react-navigation/native` (found: `6.0.10`, latest: `6.0.11`)
> 
> Can you verify that the issue still exists after upgrading to the latest versions of these packages?

still exists

**Comment by plpmd** [comment]
same is happening with me.
@expo/vector-icons": "^13.0.0",
    "@react-navigation/drawer": "^6.5.0",
    "@react-navigation/native": "^6.0.13",
    "react": "18.0.0",
    "react-native": "0.69.6",
    "react-native-gesture-handler": "~2.5.0",
    "react-native-reanimated": "~2.9.1",

**Comment by mrozkosz** [comment]
issue still exists

**Comment by esthlosx** [comment]
I have the same problem. My setup is:

    "react-native": "0.71.7",
    "@react-navigation/drawer": "^6.6.2",
    "@react-navigation/native": "^6.1.6".

**Comment by DanielCMovicoders** [comment]
Do we have any feedback on this topic? I have just installed the latest version of react-navigation/Drawer (v6.6.3) and this problem is still present. I don't remember it to be happening on version 5.

Here there are my dependencies:

```
"@react-native-community/masked-view": "^0.1.11",
"@react-navigation/drawer": "^6.6.3",
"@react-navigation/native": "^6.1.7",
"@react-navigation/native-stack": "^6.9.13",
"react": "18.2.0",
"react-native": "0.72.0",
"react-native-gesture-handler": "^2.12.0",
"react-native-reanimated": "^3.3.0",
"react-native-safe-area-context": "^4.6.3",
"react-native-screens": "^3.22.0"
```

And here you can take a look on the project
[https://github.com/DanielCMovicoders/react-native-navigator-test/tree/master](url)

**Comment by Marciomacdev** [comment]
Hi guys ! News ?

**Comment by Marciomacdev** [comment]
Hi Guys !
It worked here. 
```
   drawerStyle: {
        width: 329
   }
```
Set width property value.

**Comment by baole444** [comment]
> Hi Guys ! It worked here.
> 
> ```
>    drawerStyle: {
>         width: 329
>    }
> ```
> 
> Set width property value.

This is the fix, currently on version `@6.7.2` of Drawer.
A slightly more detail implement combine with Dimension call to return a dynamic drawer width.
```
const Drawer = createDrawerNavigator();
const windowsDimensions = Dimensions.get('window');
const screenDimensions = Dimensions.get('screen');

function App() {

    const [dimensions, setDimensions] = useState({
        window: windowsDimensions,
        screen: screenDimensions,
    });

    useEffect(() => {
      const listener = Dimensions.addEventListener(
        'change', ({window, screen}) => {
          setDimensions({window, screen});
        },
      );
      return () => listener?.remove();
    });

  return (
    <View style={styles.container}>
      <NavigationContainer>
        <Drawer.Navigator 
          screenOptions={
            {
              drawerStyle: {
                width: windowsDimensions.width * 0.5
              },
            }
          }
        >
          <Drawer.Screen name = "Your screen" component={Your_screen_function} />
        </Drawer.Navigator>
      </NavigationContainer>
  );
}
```

React Navigation's example : [drawerStyle](https://reactnavigation.org/docs/drawer-navigator/#options)
